mobile/src/main/java/org/openhab/habdroid/model/IconResource.kt

```diff
@@ -35,50 +36,81 @@ data class IconResource internal constructor(
     internal val customState: String
 ) : Parcelable {
     fun toUrl(context: Context, includeState: Boolean): String {
-        return toUrl(includeState, context.getPrefs().getIconFormat())
+        val iconSize = context.resources.getDimensionPixelSize(R.dimen.widgetlist_icon_size)
+        return toUrl(includeState, context.getPrefs().getIconFormat(), iconSize)
     }
 
     @VisibleForTesting
-    fun toUrl(includeState: Boolean, iconFormat: IconFormat): String {
+    fun toUrl(includeState: Boolean, iconFormat: IconFormat, desiredSizePixels: Int): String {
         if (!isOh2) {
             return "images/$icon.png"
         }
 
-        var iconName = "none"
+        var iconSource = "oh"
         var iconSet = "classic"
+        var iconName = "none"
 
         val segments = icon.split(":", limit = 3)
         when (segments.size) {
             1 -> iconName = segments[0]
             2 -> {
-                // Keep iconName=none for unsupported icon sources
-                if (segments[0] == "oh") {
-                    iconName = segments[1]
+                iconSource = segments[0]
+                iconName = segments[1]
+                if (iconSource == "material") {
+                    iconSet = "baseline"
                 }
             }
             3 -> {
-                // Keep iconName=none for unsupported icon sources
-                if (segments[0] == "oh") {
-                    iconSet = segments[1]
-                    iconName = segments[2]
-                }
+                iconSource = segments[0]
+                iconSet = segments[1]
+                iconName = segments[2]
             }
         }
 
-        val suffix = when (iconFormat) {
-            IconFormat.Png -> "PNG"
-            IconFormat.Svg -> "SVG"
+        when (iconSource) {
+            "material" -> {
+                iconSource = "iconify"
+                iconSet = "ic"
+                iconName = "$iconSet-$iconName"
```

Comment:
You need to swap these two lines. Otherwise the `iconName` is always `ic-$iconName`. Maybe you can extend the current tests to check all possible icon source / icon set combinations?

Reply:
oops! fixed and tests added